util/task/ticker_task.go

```diff
@@ -0,0 +1,53 @@
+package task
+
+import (
+	"time"
+)
+
+type Runner interface {
+	Run() error
+}
+
+type TickerTask struct {
+	interval time.Duration
+	runner   Runner
+	done     chan struct{}
+}
+
+func NewTickerTask(interval time.Duration, runner Runner) TickerTask {
```

Comment:
What do you think about taking a function instead of the interface ? 
So an ad-hoc closure could be passed as well, without having to define a type and method for it.